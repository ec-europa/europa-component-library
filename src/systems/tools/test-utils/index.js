const merge = require('deepmerge');
const twing = require('@ecl-twig/ec-storybook/.storybook/environment');

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) => {
    try {
      const html = twing.render(file, options);
      const test = document.createElement('test');
      test.innerHTML = html.trim();
      resolve(test);
    } catch (error) {
      reject(error);
    }
  });

module.exports = {
  merge,
  renderTwigFileAsNode,
};
