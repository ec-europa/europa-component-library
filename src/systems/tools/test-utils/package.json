{
  "name": "@ecl-twig/test-utils",
  "version": "2.31.1",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
