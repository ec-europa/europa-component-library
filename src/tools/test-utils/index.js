const merge = require('deepmerge');
const twing = require('@ecl/twig-ec-storybook/.storybook/environment');

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) => {
    try {
      const html = twing.render(file, options);
      const div = document.createElement('div');
      div.innerHTML = html.trim();
      resolve(div.firstChild);
    } catch (error) {
      reject(error);
    }
  });

module.exports = {
  merge,
  renderTwigFileAsNode,
};
