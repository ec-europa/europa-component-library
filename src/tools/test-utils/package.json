{
  "name": "@ecl/test-utils",
  "version": "3.4.2",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
