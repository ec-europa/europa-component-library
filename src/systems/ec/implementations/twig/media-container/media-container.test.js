import { merge, renderTwigFileAsNode } from '@ecl-twig/test-utils';

import demoContentImg from './demo/data--image';
import demoContentVideo from './demo/data--video';
import demoContentEmbed from './demo/data--embed-video';

describe('EC Media Container', () => {
  const template =
    '@ecl-twig/ec-component-media-container/ecl-media-container.html.twig';
  const render = (params) => renderTwigFileAsNode(template, params);
  const defaultDataStructure = demoContentImg;

  describe('Media Container generic tests', () => {
    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const optionsWithExtraClasses = merge(defaultDataStructure, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(optionsWithExtraClasses)).resolves.toMatchSnapshot();
    });

    test('Media container renders correctly with extra attributes', () => {
      expect.assertions(1);

      const optionsWithExtraClasses = merge(defaultDataStructure, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(optionsWithExtraClasses)).resolves.toMatchSnapshot();
    });
  });

  describe('EC - Media container image', () => {
    test('Media container image renders correctly', () => {
      expect.assertions(1);
      return expect(render(defaultDataStructure)).resolves.toMatchSnapshot();
    });
  });

  describe('EC - Media container video', () => {
    test('renders correctly', () => {
      const options = merge(defaultDataStructure, {
        tracks: demoContentVideo.tracks,
        sources: demoContentVideo.sources,
      });

      expect.assertions(1);
      return expect(render(options)).resolves.toMatchSnapshot();
    });

    test('with embedded media renders correctly', () => {
      expect.assertions(1);
      return expect(render(demoContentEmbed)).resolves.toMatchSnapshot();
    });
  });

  describe('with validation enabled and missing input data', () => {
    test('returns the right warning message', () => {
      expect.assertions(1);

      const dataCompliance = { ...defaultDataStructure, _compliance_: true };
      dataCompliance.image = '';

      return expect(render(dataCompliance)).resolves.toMatchSnapshot();
    });
  });
});
