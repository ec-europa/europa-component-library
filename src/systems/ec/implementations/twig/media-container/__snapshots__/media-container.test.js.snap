// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EC Media Container EC - Media container image Media container image renders correctly 1`] = `
<test>
  <figure
    class="ecl-media-container"
  >
    <img
      alt="Jean Monnet banner"
      class="ecl-media-container__media"
      src="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg"
    />
    <figcaption
      class="ecl-media-container__caption"
    >
      The European Commission has put forward ambitious yet realistic proposals for a modern EU budget. It is time for an EU budget that reflects rapid developments in innovation, the economy, the environment and geopolitics, amongst others.
    </figcaption>
  </figure>
</test>
`;

exports[`EC Media Container EC - Media container video renders correctly 1`] = `
<test>
  <figure
    class="ecl-media-container"
  >
    <video
      class="ecl-media-container__media"
      controls=""
      poster="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg"
    >
      <source
        src="https://test-videos.co.uk/vids/bigbuckbunny/mp4/h264/720/Big_Buck_Bunny_720_10s_2MB.mp4"
        type="video/mp4"
      />
      <source
        src="https://test-videos.co.uk/vids/bigbuckbunny/webm/vp9/720/Big_Buck_Bunny_720_10s_2MB.webm"
        type="video/webm"
      />
      <track
        kind="captions"
        label="English"
        src="/captions/bunny-en.vtt"
        srclang="en"
      />
      <track
        kind="captions"
        label="franÃ§ais"
        src="/captions/bunny-fr.vtt"
        srclang="fr"
      />
    </video>
    <figcaption
      class="ecl-media-container__caption"
    >
      The European Commission has put forward ambitious yet realistic proposals for a modern EU budget. It is time for an EU budget that reflects rapid developments in innovation, the economy, the environment and geopolitics, amongst others.
    </figcaption>
  </figure>
</test>
`;

exports[`EC Media Container EC - Media container video with embedded media renders correctly 1`] = `
<test>
  <figure
    class="ecl-media-container"
  >
    <div
      class="ecl-media-container__media ecl-media-container__media--ratio-16-9"
    >
      <iframe
        allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
        allowfullscreen=""
        frameborder="0"
        height="197"
        src="https://www.youtube.com/embed/fgi-GSCB6ho"
        title="New digital strategy"
        width="350"
      />
    </div>
    <figcaption
      class="ecl-media-container__caption"
    >
      The European Commission has put forward ambitious yet realistic proposals for a modern EU budget. It is time for an EU budget that reflects rapid developments in innovation, the economy, the environment and geopolitics, amongst others.
    </figcaption>
  </figure>
</test>
`;

exports[`EC Media Container Media Container generic tests Media container renders correctly with extra attributes 1`] = `
<test>
  <figure
    class="ecl-media-container"
    data-test="data-test-value"
    data-test-1="data-test-value-1"
  >
    <img
      alt="Jean Monnet banner"
      class="ecl-media-container__media"
      src="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg"
    />
    <figcaption
      class="ecl-media-container__caption"
    >
      The European Commission has put forward ambitious yet realistic proposals for a modern EU budget. It is time for an EU budget that reflects rapid developments in innovation, the economy, the environment and geopolitics, amongst others.
    </figcaption>
  </figure>
</test>
`;

exports[`EC Media Container Media Container generic tests renders correctly with extra class names 1`] = `
<test>
  <figure
    class="ecl-media-container custom-class custom-class--test"
  >
    <img
      alt="Jean Monnet banner"
      class="ecl-media-container__media"
      src="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg"
    />
    <figcaption
      class="ecl-media-container__caption"
    >
      The European Commission has put forward ambitious yet realistic proposals for a modern EU budget. It is time for an EU budget that reflects rapid developments in innovation, the economy, the environment and geopolitics, amongst others.
    </figcaption>
  </figure>
</test>
`;

exports[`EC Media Container with validation enabled and missing input data returns the right warning message 1`] = `
<test>
  <figure
    class="ecl-media-container"
  >
    <img
      alt="Jean Monnet banner"
      class="ecl-media-container__media"
      src=""
    />
    <figcaption
      class="ecl-media-container__caption"
    >
      The European Commission has put forward ambitious yet realistic proposals for a modern EU budget. It is time for an EU budget that reflects rapid developments in innovation, the economy, the environment and geopolitics, amongst others.
    </figcaption>
  </figure>
  <!-- ecl2-compliance: media-container -->
  <div
    class="ecl2-compliance ecl-u-border-width-1 ecl-u-border-color-red ecl-u-border-style-solid ecl-u-ma-xs ecl-u-pr-s ecl-u-bg-white"
  >
    <p
      class="ecl-u-type-color-grey ecl-u-ml-m ecl-u-type-xs"
    >
      [media-container] - One media is required for, an {{ image }}, an {{ embedded_media }} or a {{ video }} with {{ tracks }} and {{ sources }}.
    </p>
  </div>
</test>
`;
